googler

```diff
@@ -19,16 +19,21 @@
 
 from __future__ import print_function
 import argparse
+import atexit
 import collections
+import functools
 import gzip
 import http.client
 import html.entities
 import html.parser
 import io
+import json
+import logging
 import os
 import re
 import signal
 import sys
+import tempfile
```

Comment:
I'll step down by importing `tempfile` and `json` before they're called. Importing at argparsing time really is pretty bad, it's too hard to keep track. I just checked, even pylint can't catch unused imports at random locations within functions.